### ISSUE TITLE
failed to download modules: exec: "go"

### ISSUE THREAD

**Issue by fschuch** [benmatselby/hugo-deploy-gh-pages#38]
Hi,

I've been using this action for a while, following the [Tutorial](https://www.jameswright.xyz/post/deploy-hugo-academic-using-githubio/). But after updating to a newer version of Wowchemy, I'm getting this error:

```
Error: failed to download modules: exec: "go": executable file not found in $PATH
```

I tried to install go in many different ways, for instance:

```
    - name: setup go
      uses: actions/setup-go@v1
      with:
        go-version: 1.16.x
```

But the error persists.

Do you have any idea about how to solve it?

**Comment by benmatselby** [comment]
Hey @fschuch Can you provide some more context please. The `hugo-deploy-gh-pages` action pulls down a binary, so you shouldn't be seeing a Go issue in this action. I don't know what Wowchemy is, sadly.

**Comment by benmatselby** [comment]
@fschuch I've just completed a deployment using the `@master` tag, and it was fine.

**Comment by jrwrigh** [comment]
Wowchemy is just a Hugo theme. I haven't upgraded to the newer versions of Wowchemy recently, but I haven't run into any issues.

One thing to check @fschuch is what version of Hugo works with your version of Wowchemy. They often run off of older versions of Hugo.

**Comment by fschuch** [comment]
I appreciate your comments, @benmatselby and @jrwrigh.

Just to complement with more context, Wowchemy is a Hugo Theme. After a recent update, the workflow is depending on `go` , and I can’t tell if it is because of Hugo (maybe not, since you can run the action). I just hoped to find someone that already faced this problem. 

I’ll close the issue, since it is not directly related to the github action. I also will post the solution here, if I figure out how to solve the problem, because it may help more users.

**Comment by jrwrigh** [comment]
Just to make sure we have versions sorted out, what version of Wowchemy are you upgrading to/from? Same with the Hugo version.

I haven't made the jump to Wowchemy 5.x.x yet, so there could easily be something off with my tutorial.

**Comment by jrwrigh** [comment]
Looking [at the upgrade docs](https://wowchemy.com/blog/v5.0.0-beta.0/), they do make this mention in regards to Hugo's move to using modules:

> the full version of Hugo now requires Git and Go,

I'm not sure to what extent this is true (it maybe only needed for those moving to the new module system), but it might actually be something requiring a change to this GH action for all users. @benmatselby Do you know if your website is using the new modules system?

Also, ref [this page on Hugo's site](https://gohugo.io/hugo-modules/):

> This [Hugo modules] is all very much brand new and there are only a few example projects around

**Comment by fschuch** [comment]
Thanks for your comments, @jrwrigh.

I updated from the theme Wowchemy 4.8 with Hugo 0.74.x extended to Wowchemy 5.1 with Hugo 0.80.0 extended.

**Comment by k4rtik** [comment]
> I also will post the solution here, if I figure out how to solve the problem, because it may help more users.

@fschuch did you figure out a working solution?

**Comment by baokiemanh** [comment]
> > I also will post the solution here, if I figure out how to solve the problem, because it may help more users.
> 
> @fschuch did you figure out a working solution?

@k4rtik just follow the Go install at https://golang.org/doc/install is work for me!

**Comment by RiseAI** [comment]
> > > I also will post the solution here, if I figure out how to solve the problem, because it may help more users.
> > 
> > 
> > @fschuch did you figure out a working solution?
> 
> @k4rtik just follow the Go install at https://golang.org/doc/install is work for me!

Not sure what this means. That install doc is for installing Go locally.

**Comment by jrwrigh** [comment]
So I've tried adding the setup-go@v2 GHA to add the golang executable, but that (for some reason) didn't actually do anything. It's quite possible I'm using it incorrectly. Though

I've now forked this GHA and added `golang` to the install setup in the Docker image. I no longer get the above error message,  but this now gives out [a new error](https://github.com/jrwrigh/jameswright.xyz_source/runs/4602803565?check_suite_focus=true#step:7:22):

```
Error: failed to download modules: go command failed: go: cannot find GOROOT directory: /opt/hostedtoolcache/go/1.17.5/x64
```
This step installs go v1.11, but the output command references 1.17.5, which is what setup-go@v2 installs. Not sure what's happening, but the environment is getting wonky somewhere.
Not sure what to do from here though.

**Comment by antaldaniel** [comment]
This is where I am stuck, too.  It is unclear to me where go got installed and to what profile should I add its path.

**Comment by jrwrigh** [comment]
I believe I fixed it. I have no idea why https://github.com/actions/setup-go doesn't seem to work here, but apparently my issue before was having `setup-go` and the docker install `golang`. Removing `setup-go` and only adding `golang` to the docker image solved the issue for me. 

@antaldaniel See if it works for you. Simply apply

```diff
diff --git a/.github/workflows/BuildandDeploy.yml b/.github/workflows/BuildandDeploy.yml
index 3673cb9..8ce203f 100644
--- a/.github/workflows/BuildandDeploy.yml
+++ b/.github/workflows/BuildandDeploy.yml
@@ -23,7 +23,8 @@ jobs:
         submodules: true
 
     - name: Hugo Deploy GitHub Pages
-      uses: benmatselby/hugo-deploy-gh-pages@master
+      uses: jrwrigh/hugo-deploy-gh-pages@fix/missing_go_exec
       env:
         HUGO_VERSION: 0.89.4
         HUGO_EXTENDED: true
```

to your workflows yaml file.

**Comment by benmatselby** [comment]
Hey folks. I've just released 1.13.0 which allows you to define a `GO_VERSION` in the environment variables. Please check out the README. Hope this helps your use case.
